gio/src/subclass/icon.rs

```diff
@@ -0,0 +1,101 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::Icon;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::{Cast, IsA};
+
+pub trait IconImpl: ObjectImpl {
+    fn hash(&self, icon: &Self::Type) -> u32;
+    fn equal(&self, icon: &Self::Type, other_icon: &Icon) -> bool;
```

Comment:
Or require `Hash`, `PartialEq` and `Eq` to be implemented?

Reply:
Much better indeed.